moveit_simple/include/moveit_simple/moveit_simple.h

```diff
@@ -352,4 +360,47 @@ class CartTrajectoryPoint : public TrajectoryPoint
   Eigen::Affine3d pose_;
 };
 
+class addTrajPointException: public std::runtime_error
```

Comment:
Yes.  Take the standard library for example.  The exception hierarchy is defined outside of any of the standard libraries/templates.  It's because failure modes are not unique to any function/class, but rather are cross cutting.  In this case, I expect it will result in far fewer exceptions.  It's easier to code, easier to document, and easier for others to use.